beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -2086,6 +2214,17 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
             }
         }
 
+        // Apply the state to the attester cache, only if it is from the previous epoch or earlier.
```

Comment:
This seems mismatched with the condition, which adds states with `epochs >= current_epoch - 2`, i.e. newer than the previous previous epoch.